kernel/src/table_features/mod.rs

```diff
@@ -124,6 +150,424 @@ pub(crate) enum FeatureType {
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::NotSupported,
+    enablement_check: EnablementCheck::EnabledIf(|props| {
+        props.enable_change_data_feed == Some(true)
+    }),
+};
+
+#[allow(dead_code)]
+static GENERATED_COLUMNS_INFO: FeatureInfo = FeatureInfo {
+    name: "generatedColumns",
+    min_reader_version: 1,
+    min_writer_version: 4,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::NotSupported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static IDENTITY_COLUMNS_INFO: FeatureInfo = FeatureInfo {
+    name: "identityColumns",
+    min_reader_version: 1,
+    min_writer_version: 6,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::NotSupported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static IN_COMMIT_TIMESTAMP_INFO: FeatureInfo = FeatureInfo {
+    name: "inCommitTimestamp",
+    min_reader_version: 3,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::Supported,
+    enablement_check: EnablementCheck::EnabledIf(|props| {
+        props.enable_in_commit_timestamps == Some(true)
+    }),
+};
+
+#[allow(dead_code)]
+static ROW_TRACKING_INFO: FeatureInfo = FeatureInfo {
+    name: "rowTracking",
+    min_reader_version: 3,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[FeatureRequirement::Supported(TableFeature::DomainMetadata)],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::Supported,
+    enablement_check: EnablementCheck::EnabledIf(|props| {
+        props.enable_row_tracking == Some(true) && props.row_tracking_suspended != Some(true)
+    }),
+};
+
+#[allow(dead_code)]
+static DOMAIN_METADATA_INFO: FeatureInfo = FeatureInfo {
+    name: "domainMetadata",
+    min_reader_version: 3,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::Supported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+// TODO(#1125): IcebergCompatV1 requires schema type validation to block Map, Array, and Void types.
+// This validation is not yet implemented. The feature is marked as NotSupported for writes until proper validation is added.
+// See Delta Spark: IcebergCompat.scala CheckNoListMapNullType (lines 422-433)
+// See Java Kernel: IcebergWriterCompatMetadataValidatorAndUpdater.java UNSUPPORTED_TYPES_CHECK
+#[allow(dead_code)]
+static ICEBERG_COMPAT_V1_INFO: FeatureInfo = FeatureInfo {
```

Comment:
There's a bunch more stuff that needs to be verified for iceberg compat (like DVs being off and purged for instance). I think it's great to have this in here and we can just keep it marked as unsupported, but we should probably just note that there is work to be done to finish the compatibility checks beyond just the type checks. Ditto for V2 which has even more stuff